utilities/cli/holohub.py

```diff
@@ -259,40 +259,55 @@ def _find_project(self, project_name: str, language: Optional[str] = None) -> di
         return None
 
     def _make_project_container(
-        self, project_name: Optional[str] = None, language: Optional[str] = None
+        self,
+        project_name: Optional[str] = None,
+        language: Optional[str] = None,
+        dryrun: bool = False,
+        build_args: Optional[dict] = None,
     ) -> HoloHubContainer:
-        """Define a project container"""
+        """Define a project container and optionally build it
+
+        Args:
+            project_name: Name of the project
+            language: Language implementation (cpp/python)
+            dryrun: Whether to run in dryrun mode
+            build_args: Optional arguments to pass to container.build()
+        """
         if not project_name:
-            return HoloHubContainer(project_metadata=None)
+            container = HoloHubContainer(project_metadata=None)
+        else:
+            project_data = self._find_project(project_name=project_name, language=language)
+            container = HoloHubContainer(project_metadata=project_data)
 
-        project_data = self._find_project(project_name=project_name, language=language)
-        if not project_data:
-            holohub_cli_util.fatal(
-                f"Project '{project_name}' {f'({language})' if language else ''} not found"
-            )
-        return HoloHubContainer(project_metadata=project_data)
+        container.dryrun = dryrun
+        if build_args is not None:
+            container.build(**build_args)
```

Comment:
I wonder about this heuristic change. Is there a clear motivation for changing to implicitly build, rather than always calling `container.build` explicitly on the resulting object?